rest/system_metrics_routes_test.go

```diff
@@ -0,0 +1,184 @@
+
+	suite.Suite
+}
+
+func (s *systemMetricsHandlerSuite) setup(tempDir string) {
+	s.sc = data.MockConnector{
+		Bucket: tempDir,
+		CachedSystemMetrics: map[string]dbModel.SystemMetrics{
+			"abc": dbModel.SystemMetrics{
+				ID: "abc",
+				Info: dbModel.SystemMetricsInfo{
+					Project:   "test",
+					Version:   "0",
+					Variant:   "linux",
+					TaskName:  "task0",
+					TaskID:    "task1",
+					Execution: 0,
+					Mainline:  true,
+				},
+				Artifact: dbModel.SystemMetricsArtifactInfo{
+					Prefix: "abc",
+					Options: dbModel.SystemMetricsArtifactOptions{
+						Type: dbModel.PailLocal,
+					},
+				},
+			},
+			"def": dbModel.SystemMetrics{
+				ID: "def",
+				Info: dbModel.SystemMetricsInfo{
+					Project:   "test",
+					Version:   "0",
+					Variant:   "linux",
+					TaskName:  "task0",
+					TaskID:    "task1",
+					Execution: 1,
+					Mainline:  true,
+				},
+				Artifact: dbModel.SystemMetricsArtifactInfo{
+					Prefix: "def",
+					Options: dbModel.SystemMetricsArtifactOptions{
+						Type: dbModel.PailLocal,
+					},
+				},
+			},
+			"ghi": dbModel.SystemMetrics{
+				ID: "ghi",
+				Info: dbModel.SystemMetricsInfo{
+					Project:   "test",
+					Version:   "0",
+					Variant:   "linux",
+					TaskName:  "task2",
+					TaskID:    "task3",
+					Execution: 0,
+					Mainline:  true,
+				},
+				Artifact: dbModel.SystemMetricsArtifactInfo{
+					Prefix: "ghi",
+					Options: dbModel.SystemMetricsArtifactOptions{
+						Type: dbModel.PailLocal,
+					},
+				},
+			},
+		},
+	}
+
+	for key, val := range s.sc.CachedSystemMetrics {
+		val.Artifact.MetricChunks = map[string]dbModel.MetricChunks{
+			"uptime": dbModel.MetricChunks{
+				Chunks: []string{fmt.Sprintf("%v", time.Now())},
+				Format: dbModel.FileText,
+			},
+		}
+		s.sc.CachedSystemMetrics[key] = val
+
+		opts := pail.LocalOptions{
+			Path:   s.sc.Bucket,
+			Prefix: val.Artifact.Prefix,
+		}
+		bucket, err := pail.NewLocalBucket(opts)
+		s.Require().NoError(err)
+		data := []byte(fmt.Sprintf("%s-%d", val.Info.TaskID, val.Info.Execution))
+		s.Require().NoError(bucket.Put(context.TODO(), val.Artifact.MetricChunks["uptime"].Chunks[0], bytes.NewReader(data)))
+	}
+
+	s.rh = map[string]gimlet.RouteHandler{
+		"type": makeGetSystemMetricsByType(&s.sc),
+	}
+
+}
+
+func TestSystemMetricsHandlerSuite(t *testing.T) {
+	s := new(systemMetricsHandlerSuite)
+	tempDir, err := ioutil.TempDir(".", "system-metrics")
+	s.Require().NoError(err)
+	defer func() {
+		s.NoError(os.RemoveAll(tempDir))
+	}()
+	s.setup(tempDir)
+	suite.Run(t, s)
+}
+
+func (s *systemMetricsHandlerSuite) TestGetSystemMetricsByTypeFound() {
+	// no execution
+	rh := s.rh["type"]
+	rh.(*systemMetricsGetByTypeHandler).opts.TaskID = "task1"
+	rh.(*systemMetricsGetByTypeHandler).opts.EmptyExecution = true
+	rh.(*systemMetricsGetByTypeHandler).metricType = "uptime"
+
+	resp := rh.Run(context.TODO())
+	s.Require().NotNil(resp)
+	s.Equal(http.StatusOK, resp.Status())
+	r, ok := resp.Data().(io.ReadCloser)
```

Comment:
I find this test to be a bit odd - if you're serializing the payload as JSON and sending it over the network, won't the end user receive nothing?